### ISSUE TITLE
SharpKeys throws after unminimizing

### ISSUE THREAD

**Issue by wenfangdu** [randyrants/sharpkeys#445]
SharpKeys throws after unminimizing

Repro steps:

1. Minimize the window
2. Alt + Tab to bring it back
3. Window throws an error

![image](https://user-images.githubusercontent.com/28700378/161204543-c039b304-f1a7-44a6-aa2c-375562fabdfc.png)

Log:

```
有关调用实时(JIT)调试而不是此对话框的详细信息，
请参见此消息的结尾。

************** 异常文本 **************
System.ArgumentException: 矩形“{X=0,Y=0,Width=0,Height=0}”的宽度或高度不能等于零。
   在 System.Drawing.Drawing2D.LinearGradientBrush..ctor(Rectangle rect, Color color1, Color color2, LinearGradientMode linearGradientMode)
   在 SharpKeys.Dialog_Main.mainPanel_Paint(Object sender, PaintEventArgs e)
   在 System.Windows.Forms.Control.OnPaint(PaintEventArgs e)
   在 System.Windows.Forms.Control.PaintTransparentBackground(PaintEventArgs e, Rectangle rectangle, Region transparentRegion)
   在 System.Windows.Forms.Control.PaintBackground(PaintEventArgs e, Rectangle rectangle, Color backColor, Point scrollOffset)
   在 System.Windows.Forms.Control.PaintBackground(PaintEventArgs e, Rectangle rectangle)
   在 System.Windows.Forms.Control.OnPaintBackground(PaintEventArgs pevent)
   在 System.Windows.Forms.ScrollableControl.OnPaintBackground(PaintEventArgs e)
   在 System.Windows.Forms.Control.PaintWithErrorHandling(PaintEventArgs e, Int16 layer)
   在 System.Windows.Forms.Control.WmEraseBkgnd(Message& m)
   在 System.Windows.Forms.Control.WndProc(Message& m)
   在 System.Windows.Forms.NativeWindow.Callback(IntPtr hWnd, Int32 msg, IntPtr wparam, IntPtr lparam)


************** 已加载的程序集 **************
mscorlib
    程序集版本:4.0.0.0
    Win32 版本:4.8.4470.0 built by: NET48REL1LAST_C
    基本代码:file:///C:/Windows/Microsoft.NET/Framework64/v4.0.30319/mscorlib.dll
----------------------------------------
SharpKeys
    程序集版本:3.9.3.0
    Win32 版本:3.9.3.0
    基本代码:file:///C:/Program%20Files/RandyRants.com/SharpKeys/SharpKeys.exe
----------------------------------------
System.Windows.Forms
    程序集版本:4.0.0.0
    Win32 版本:4.8.4488.0 built by: NET48REL1LAST_C
    基本代码:file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/System.Windows.Forms/v4.0_4.0.0.0__b77a5c561934e089/System.Windows.Forms.dll
----------------------------------------
System
    程序集版本:4.0.0.0
    Win32 版本:4.8.4488.0 built by: NET48REL1LAST_C
    基本代码:file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/System/v4.0_4.0.0.0__b77a5c561934e089/System.dll
----------------------------------------
System.Drawing
    程序集版本:4.0.0.0
    Win32 版本:4.8.4390.0 built by: NET48REL1LAST_C
    基本代码:file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/System.Drawing/v4.0_4.0.0.0__b03f5f7f11d50a3a/System.Drawing.dll
----------------------------------------
System.Configuration
    程序集版本:4.0.0.0
    Win32 版本:4.8.4190.0 built by: NET48REL1LAST_B
    基本代码:file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/System.Configuration/v4.0_4.0.0.0__b03f5f7f11d50a3a/System.Configuration.dll
----------------------------------------
System.Core
    程序集版本:4.0.0.0
    Win32 版本:4.8.4470.0 built by: NET48REL1LAST_C
    基本代码:file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/System.Core/v4.0_4.0.0.0__b77a5c561934e089/System.Core.dll
----------------------------------------
System.Xml
    程序集版本:4.0.0.0
    Win32 版本:4.8.4084.0 built by: NET48REL1
    基本代码:file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/System.Xml/v4.0_4.0.0.0__b77a5c561934e089/System.Xml.dll
----------------------------------------
System.Drawing.resources
    程序集版本:4.0.0.0
    Win32 版本:4.8.4084.0 built by: NET48REL1
    基本代码:file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/System.Drawing.resources/v4.0_4.0.0.0_zh-Hans_b03f5f7f11d50a3a/System.Drawing.resources.dll
----------------------------------------
System.Windows.Forms.resources
    程序集版本:4.0.0.0
    Win32 版本:4.8.4084.0 built by: NET48REL1
    基本代码:file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/System.Windows.Forms.resources/v4.0_4.0.0.0_zh-Hans_b77a5c561934e089/System.Windows.Forms.resources.dll
----------------------------------------
mscorlib.resources
    程序集版本:4.0.0.0
    Win32 版本:4.8.4084.0 built by: NET48REL1
    基本代码:file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/mscorlib.resources/v4.0_4.0.0.0_zh-Hans_b77a5c561934e089/mscorlib.resources.dll
----------------------------------------

************** JIT 调试 **************
要启用实时(JIT)调试，
该应用程序或计算机的 .config 文件(machine.config)的 system.windows.forms 节中必须设置
jitDebugging 值。
编译应用程序时还必须启用
调试。
```

**Comment by randyrants** [comment]
Oof: it's even worse than that.  Any restore from a minimized state causes the crash, but given where it's coming from, I'm wondering if the .NET Framework has updated and has a regression (or possibly even a "fix" since it looks like it's now checking for an "invisible" window.

I'll have to investigate if there's a workaround. 

Thanks for reporting this!

**Comment by randyrants** [comment]
This has been fixed in 3.9.4 - please let me know if it's not cleared up for you.
